apps/builder/app/builder/features/settings-panel/props-section/animation/animation-keyframes.tsx

```diff
@@ -0,0 +1,218 @@
+      styleSource="default"
+      /* same as offset has 0 - 100% */
+      property={"fontStretch"}
+      value={
+        value !== undefined
+          ? {
+              type: "unit",
+              value: Math.round(value * 1000) / 10,
+              unit: "%",
+            }
+          : undefined
+      }
+      onChange={(styleValue) => {
+        if (styleValue === undefined) {
+          setIntermediateValue(styleValue);
+          return;
+        }
+
+        const clampedStyleValue = { ...styleValue };
+        if (
+          clampedStyleValue.type === "unit" &&
+          clampedStyleValue.unit === "%"
+        ) {
+          clampedStyleValue.value = Math.min(
+            100,
+            Math.max(0, clampedStyleValue.value)
+          );
+        }
+
+        setIntermediateValue(clampedStyleValue);
+      }}
+      onHighlight={(_styleValue) => {
+        /* @todo: think about preview */
+      }}
+      onChangeComplete={(event) => {
+        setIntermediateValue(undefined);
+
+        if (event.value.type === "unit" && event.value.unit === "%") {
+          onChange(Math.min(100, Math.max(0, event.value.value)) / 100);
+          return;
+        }
+
+        setIntermediateValue({
+          type: "invalid",
+          value: toValue(event.value),
+        });
+      }}
+      onAbort={() => {
+        /* @todo: allow to change some ephemeral property to see the result in action */
+      }}
+      onReset={() => {
+        setIntermediateValue(undefined);
+        onChange(undefined);
+      }}
+    />
+  );
+};
+
+const Keyframe = ({
+  value,
+  onChange,
+}: {
+  value: AnimationKeyframe;
+  onChange: (value: AnimationKeyframe | undefined) => void;
+}) => {
+  const ids = useIds(["offset"]);
+
+  const cssProperties = useMemo(() => {
+    let result = ``;
+    for (const [property, style] of Object.entries(value.styles)) {
+      result = `${result}${toKebabCase(property)}: ${toValue(style)};\n`;
+    }
+    return result;
+  }, [value.styles]);
+
+  return (
+    <>
+      <Grid
+        gap={1}
+        align={"center"}
+        css={{ gridTemplateColumns: "1fr 1fr auto" }}
+      >
+        <Label htmlFor={ids.offset}>Offset</Label>
+        <OffsetInput
+          id={ids.offset}
+          value={value.offset}
+          onChange={(offset) => {
+            onChange({ ...value, offset });
+          }}
+        />
+        <Tooltip content="Remove keyframe">
+          <IconButton onClick={() => onChange(undefined)}>
+            <MinusIcon />
+          </IconButton>
+        </Tooltip>
+      </Grid>
+      <Grid>
+        <CodeEditor
+          lang="css-properties"
+          size="keyframe"
+          value={cssProperties}
+          onChange={() => {
+            /* do nothing */
+          }}
+          onChangeComplete={(cssText) => {
+            const parsedStyles = parseCss(`selector{${cssText}}`);
+            onChange({
+              ...value,
+              styles: parsedStyles.reduce(
```

Comment:
if you rebase properties will be hyphenated like background-color instead of backgroundColor

Reply:
not expected, fixed ;)